test/helpers/utils.go

```diff
@@ -44,24 +44,47 @@ func WaitForConfigAndPoolComplete(t *testing.T, cs *framework.ClientSet, pool, m
 // included the given mcName in its config, and returns the new
 // rendered config name.
 func WaitForRenderedConfig(t *testing.T, cs *framework.ClientSet, pool, mcName string) (string, error) {
+	return WaitForRenderedConfigs(t, cs, pool, mcName)
+}
+
+// WaitForRenderedConfigs polls a MachineConfigPool until it has
+// included the given mcNames in its config, and returns the new
+// rendered config name.
+func WaitForRenderedConfigs(t *testing.T, cs *framework.ClientSet, pool string, mcNames ...string) (string, error) {
 	var renderedConfig string
 	startTime := time.Now()
 	if err := wait.PollImmediate(2*time.Second, 5*time.Minute, func() (bool, error) {
+		// Set up the list
+		found := make(map[string]bool)
+		for _, name := range mcNames {
+			found[name] = false
+		}
+
+		// Update found based on the MCP
 		mcp, err := cs.MachineConfigPools().Get(context.TODO(), pool, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
 		for _, mc := range mcp.Spec.Configuration.Source {
-			if mc.Name == mcName {
-				renderedConfig = mcp.Spec.Configuration.Name
-				return true, nil
+			if _, ok := found[mc.Name]; ok {
+				found[mc.Name] = true
 			}
 		}
-		return false, nil
+
+		// If any are still false, then they weren't included in the MCP
+		for _, nameFound := range found {
+			if !nameFound {
+				return false, nil
+			}
+		}
+
+		// All the required names were found
+		renderedConfig = mcp.Spec.Configuration.Name
+		return true, nil
 	}); err != nil {
-		return "", errors.Wrapf(err, "machine config %s hasn't been picked by pool %s (waited %s)", mcName, pool, time.Since(startTime))
+		return "", errors.Wrapf(err, "machine configs %v hasn't been picked by pool %s (waited %s)", mcNames, pool, time.Since(startTime))
```

Comment:
Since we are applying multiple MCs, it is possible that few MCs got applied and few didn't. For better debugging, it would make sense to log which MCs wasn't got picked up instead of logging all MCs.